src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -168,21 +171,103 @@ trait EntityComponent {
       allEntitiesAction.flatMap(cloneEntities(destWorkspaceContext, _))
     }
 
+    def insertEntitiesSql(destWorkspaceContext: SlickWorkspaceContext, entities: Map[Entity, UUID]) = {
+      val workspaceId = destWorkspaceContext.workspaceId
+
+      val insertBatches = entities.map { case (entity, entityId) =>
+        sql"($entityId, ${entity.name}, ${entity.entityType}, $workspaceId)"
+      }.grouped(1000).toSeq
+
+      DBIO.seq(insertBatches.map { insertBatch =>
+        val prefix = sql"insert into ENTITY (id, name, entity_type, workspace_id) values "
+        val suffix = insertBatch.reduce { (a, b) =>
+          concatSqlActionsWithDelim(a, b, ",")
+        }
+        concatSqlActions(prefix, suffix).as[Int]
+      }:_*)
+    }
+
+    def insertAttributesSql(destWorkspaceContext: SlickWorkspaceContext, attrsWithEntityIdMap: Seq[(UUID, (String, Attribute), UUID, Option[Int])]) = {
```

Comment:
I didn't see a generic marshalAttribute, so I didn't mess with this too much other than moving it to AttributeComponent. AttributeRecords don't line up nicely with the raw sql because of requiring NULL instead of Nones for missing values so I'd end up needing the case logic in there anyway to know where to put the null values